### PR TITLE
PythonCheck: check for missing scm dep when needed

### PR DESCRIPTION
**Pull request by arthurzam**
Resolves: https://github.com/pkgcore/pkgcheck/issues/537

### CODE REVIEW

File: src/pkgcheck/checks/python.py [hunk 1]
```diff
@@ -222,6 +223,20 @@ def desc(self):
         return f"{self.dep_category}: missing check for {self.dep_atom}[{use_flags}] in {self.location!r}"
 
 
+class PythonMissingSCMDependency(results.VersionResult, results.Warning):
+    """Package is missing BDEPEND on setuptools_scm or alike.
+
+    Packages which define ``SETUPTOOLS_SCM_PRETEND_VERSION`` should BDEPEND
+    on ``dev-python/setuptools_scm`` or similar package [#]_.
```
Comment (thesamesam) [inline src/pkgcheck/checks/python.py:230]:
```suggestion
    on ``dev-python/setuptools_scm`` or a similar package [#]_.
```

**Review by thesamesam — COMMENTED** [review]
Thanks, I like it.

File: src/pkgcheck/checks/python.py [hunk 2]
```diff
@@ -263,6 +279,14 @@ class PythonCheck(Check):
         "python-r1": "python_gen_any_dep",
     }
 
+    setuptool_scm = frozenset(
```
Comment (mgorny) [inline src/pkgcheck/checks/python.py:282]:
```suggestion
    setuptools_scm = frozenset(
```

It looks weird to make it singular.

**Review by mgorny — APPROVED** [review]
Just minor naming suggestions from me.

**Review by thesamesam — APPROVED** [review]

**Pull request merged**